serverside/src/lesson/lesson.service.ts

```diff
@@ -1,4 +1,38 @@
 import { Injectable } from '@nestjs/common';
+import { PrismaClient, Lesson } from '@prisma/client';
 
 @Injectable()
-export class LessonService {}
+export class LessonService {
+  async getLessonsForWeek(
+    weekStart: Date,
+    classID: number,
+    prisma: PrismaClient,
+  ): Promise<Lesson[][]> {
+    const endOfWeek = new Date(weekStart);
+    endOfWeek.setDate(weekStart.getDate() + 5); // Nur Montag bis Freitag, da am Wochenende keine Schule
+
+    const lessons: Lesson[] = await prisma.lesson.findMany({
+      where: {
+        classClassID: classID,
+        startTime: {
+          gte: weekStart,
+          lte: endOfWeek,
+        },
+      },
+    });
+
+    const lessonsByDay: Lesson[][] = [[], [], [], [], []];
```

Comment:
funktioniert das so? also den typ den du hier angibst ist ein zwei spalten array, aber du initialisierst das als ein array, in dem wieder 5 arrays sind

Reply:
Yes, ich lege so die Länge 5 für die Tage fest aber die Tage an sich sich unbegrenzt.